users/models.py

```diff
@@ -18,7 +16,10 @@ class User(AbstractUser):
 
     def send_verification_email(self, token):
         context = {
-            'verification_url': build_email_verification_url(self.email, token),
+            'base': settings.FRONT_END_HOST,
+            'endpoint': 'verify',
+            'email': self.email,
+            'token': token,
```

Comment:
Personally I like building the verification_url outside of the template rather than inside the template. 1) Less logic inside of the template is always good. 2) The pieces of the url have to travel around together. Why not just make one thing, then url, which binds the pieces together and you thereafter only have to worry about a single object, not 4. Consider how complicated this would be if you had 3 urls. You'd have to ship around lots of variables and makes sure that the template put them together correct.

Reply:
That's fair! How about I just change build_url to be a non-template tag? That way we still have a method that can be used in lots of different circumstances, but we don't have to pass as much around into the template.

Reply:
That sounds fine to me. In that case this code would look similar to where it started, but with a more general `build_url` method and we're back to having 2 context vars `verification_url` and `first_name`, right?

Reply:
Yup!